include/deal.II/grid/tria_levels.h

```diff
@@ -206,6 +207,13 @@ namespace internal
        */
       std::vector<unsigned char> face_orientations;
 
+      /**
+       * Geometric entity type of each cell.
+       *
+       * @note Used only for dim=2 and dim=3.
+       */
+      std::vector<ReferenceCell::Type> entity_type;
```

Comment:
I would prefer to rename this to something like `reference_cell_types` so that we are using clearer terminology than 'entity'.